### PR TITLE
PBA mask revamp

### PR DESCRIPTION
**Pull request by Maskrow**
#### Summary

SUMMARY: Features "Changed the gas mask filters."
#### Purpose of change

More realistic approach to the different types of respirators, as well as the first step to make a tiered system of air filtration equipment at the player's disposal.
#### Describe the solution

Nytrox cylinders are a magazine type item that are used up by a PBA mask while it is active. Their capacity is balanced around IRL capacity
#### Describe alternatives you've considered

I considered adding nitrox as a separate type of ammunition to allow partial reloads, but unrechargeable cylinders modeled after the acytelene torch welding tanks seem to be more in line with general atmosphere of post-apocalyptic disaster, making the player scavenge for them.
#### Additional context

There are further plans to rebalance different gas masks according to their method of supplying breathable air to the character as well as separating them into different tiers of efficiency/rarity/sustainability.

### CODE REVIEW

**Comment by Maskrow** [comment]
Jenkins rebuild

**Comment by nexusmrsep** [comment]
Nice addition overall **IF** it's an alternative to filter type gas masks. Gas masks aka filter masks have no own oxy supply but instead filter outside air to get rid of contaminants. Contaminants clog the filter, slowly rendering it unusable. There is nothing wrong with that mechanic. They should not be replaced by this kind of masks. Masks using oxy/nitrox should have a different name then "gas mask" to put them aside as a different kind of mask. They also should allow to breathe underwater using  oxy/nitrox. Question: how is this related to existing oxy tanks (large and small)? Perhaps you could merge them together?

**Comment by Maskrow** [comment]
@nexusmrsep 
The endgoal is not to remove current gas masks, but to separate different breathing equipment into different tiers that are going to have different downsides and upsides.
**Tier 1** would be simple clothing items that provide air filtration mostly as a side effect - bandanas, kuffieh, nomad cowls and such. They require no maintenance.
**Tier 2** will have simple mechanical filters - dust masks, water-soaked rags, filter masks. They will provide meager environmental protection, but are cheap as you can easily "renew" them by washing the filtration medium or crafting a new one.
**Tier 3** is the chemical filters like gas masks. They give good enough defence and their "ammunition" is plentiful and craftable (high enough skills provided).
**Tier 4** will see addition of powered air-purifying respirators (PARPs). They will provide borderline immunity to the most contaminants barring the strongest ones and will use the same filters as the gas masks. This is offset by requiring some batteries to function and being noisy.
**Tier 5** is self-contained breathing appartuses (SCBAs) like the PBA mask in this commit. They give complete immunity to outside contaminants and are low on encumberment (while active). They, however, will consume non-craftable nitrox cylinders (I don't see survivours being able to craft correct and thus safe to use nitrox mixtures easily).
**Tier 6** is power armor which is win.

As for the oxy tanks, pure oxygen can be toxic when consumed in vast quantities, thus requiring addition of separate nitrox cylinders commonly used in water- and industrial diving. But I could see adding and using different medical masks and cannulas as a substitution for different tiers, as well as adding makeshift versions.

File: data/json/items/magazine/nitrox.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
+[
+  {
+    "id": "nitrox_tank_metal",
+    "type": "MAGAZINE",
+    "name": "aluminum nitrox cylinder",
+    "description": "A standard 30 minute aluminum 2216 PSI cylinder for use with a breathing apparatus.",
+    "weight": 6000,
+    "volume": 7,
+    "price": 5600,
+    "material": "superalloy",
```
Comment (Night-Pryanik) [inline data/json/items/magazine/nitrox.json]: Its name suggests it's made from aluminum?

File: data/json/items/magazine/nitrox.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+[
+  {
+    "id": "nitrox_tank_metal",
+    "type": "MAGAZINE",
+    "name": "aluminum nitrox cylinder",
+    "description": "A standard 30 minute aluminum 2216 PSI cylinder for use with a breathing apparatus.",
+    "weight": 6000,
+    "volume": 7,
```
Comment (Night-Pryanik) [inline data/json/items/magazine/nitrox.json]: Is it really has volume of only 1750 ml?

File: data/json/items/tool_armor.json [hunk 3]
```diff
@@ -844,25 +844,62 @@
...
         "material": [ "plastic", "nomex" ],
         "weight": 1260,
         "volume": 6,
         "to_hit": -3,
-        "max_charges": 100,
-        "initial_charges": 100,
-        "ammo": "gasfilter_m",
-        "use_action": "GASMASK",
+        "ammo": "nitrox",
+		"magazines": [ [ "nitrox", [ "nitrox_tank_carbon", "nitrox_tank_metal" ] ] ],
```
Comment (Night-Pryanik) [inline data/json/items/tool_armor.json]: Tab.

File: data/json/items/tool_armor.json [hunk 4]
```diff
@@ -844,25 +844,62 @@
...
+        "description": "A full-face mask commonly used by firefighters.  The protective breathing apparatus provides excellent protection from smoke, flame, and other dangers.  It is currently active.",
+        "flags": [ "VARSIZE", "WATERPROOF", "STURDY", "SUN_GLASSES" ],
+        "price": 100000,
+        "material": [ "plastic", "nomex" ],
+        "weight": 1260,
+        "volume": 6,
+        "to_hit": -3,
+        "turns_per_charge": 3,
+        "ammo": "nitrox",
+		"magazines": [ [ "nitrox", [ "nitrox_tank_carbon", "nitrox_tank_metal" ] ] ],
```
Comment (Night-Pryanik) [inline data/json/items/tool_armor.json]: Tab.

File: data/json/items/magazine/nitrox.json (hunk 2 above)
Comment (Maskrow) [reply data/json/items/magazine/nitrox.json]: Changed volume to 47.

File: data/json/items/magazine/nitrox.json (hunk 1 above)
Comment (Maskrow) [reply data/json/items/magazine/nitrox.json]: Changed to aluminum.

**Comment by Vasyan2006** [comment]
I was told to avoid making new itemID for active items and make iuse function instead. You can add "SCBA" flag to new masks and modify "gasmask" function or make new one based on it.

Is it fine to strap 11.75L/6kg air tank to the mask? Commonly such tanks are strapped to the torso. It will be better to absorb air from any tank in players inventory (lets say they all are magically connected to the mask) if the mask is active.

To reload air tanks you basically need only an air compressor if these tanks will be used at sea level. Nitrox and other breathing mixtures are required only for breathing at low or high pressure, like flying or diving. It is strictly recommended to use air filter after compressor to prevent oil and air dust from entering the tank and to sterilize tanks after every use, but these parts can be simplified ingame.

**Comment by Maskrow** [comment]
@Vasyan2006 
I would have really liked to implement the cylinders requiring to fit them into a holster-type torso item in the "strapped" layer and their contents consumed by the mask automatically, but it seems to be outside the scope of my current skills.
Reloading the tanks with a compressor is also something I would like to add when I have time, because the power consumption.
Ok, if making this change would be much better by implementing the iuse function, then I'll proceed to fiddling with C++.

**Pull request closed without merging**